src/Progress/CircularProgress.js

```diff
@@ -53,18 +67,40 @@ export const styleSheet = createStyleSheet('MuiCircularProgress', theme => ({
 }));
 
 function CircularProgress(props) {
-  const { classes, className, size, ...other } = props;
+  const { classes, className, size, mode, value, min, max, ...other } = props;
   const radius = size / 2;
+  const rootProps = {};
+  const svgClasses = classNames(classes.svg, {
+    [classes.indeterminateSvg]: mode === 'indeterminate',
+  });
+
+  const circleClasses = classNames(classes.circle, {
+    [classes.indeterminateCircle]: mode === 'indeterminate',
+    [classes.determinateCircle]: mode === 'determinate',
+  });
+
+  const circleStyle = { };
+  if (mode === 'determinate') {
+    const relVal = getRelativeValue(value, min, max);
+    circleStyle.strokeDasharray = `calc(((100% - ${THICKNESS}px) * ${PI}) * ${relVal}),` +
+                                  `calc((100% - ${THICKNESS}px) * ${PI})`;
```

Comment:
Can you try to rebase on `next`? I'm not sure prettier is going to like that formatting. Thanks!